test/e2e/node/crictl.go

```diff
@@ -38,37 +38,25 @@ var _ = SIGDescribe("crictl", func() {
 	})
 
 	ginkgo.It("should be able to run crictl on the node", func() {
-		// Get all nodes' external IPs.
-		ginkgo.By("Getting all nodes' SSH-able IP addresses")
-		hosts, err := e2essh.NodeSSHHosts(f.ClientSet)
-		if err != nil {
-			framework.Failf("Error getting node hostnames: %v", err)
-		}
-		e2eskipper.SkipUnlessAtLeast(len(hosts), 1, "Skipping test because found no SSH'able node")
+		nodes, err := e2enode.GetBoundedReadySchedulableNodes(f.ClientSet, maxNodes)
```

Comment:
it used to skip the test if there are no ready ssh-able nodes, with this change it will fail. Is it intentional? (PR introduced test skipping: https://github.com/kubernetes/kubernetes/pull/101866)

Reply:
This new combination of `GetBoundedReadySchedulableNodes()` and `hostExec.Execute()` creates a pod on the specific node and run the specified command in the pod without using ssh.
This pr requires just schedulable nodes, and I think it is good that the test fails if there is not such nodes on the environment.

Reply:
Yeah the test is also still covered by the e2eskipper constraint:

```
ginkgo.BeforeEach(func() {
		// `crictl` is not available on all cloud providers.
		e2eskipper.SkipUnlessProviderIs("gce", "gke")
	})
```

Reply:
where is `maxNodes`coming from?

Reply:
Is there a relation of `gce,gke`and `crictl` or just the CI combination?
I think that this will run now everywhere where the binary is available, i.e., it will run on kind jobs too, 

TODO to skip if binary doesn't exist and not skip by cloud provider